hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/ReaderContext.java

```diff
@@ -39,15 +39,17 @@
   private final HFileSystem hfs;
   private final boolean primaryReplicaReader;
   private final ReaderType type;
+  private final boolean prefetchOnOpen;
 
   public ReaderContext(Path filePath, FSDataInputStreamWrapper fsdis, long fileSize,
-      HFileSystem hfs, boolean primaryReplicaReader, ReaderType type) {
```

Comment:
Just because the PreadREader uses the Context you are adding it here and StorefileInfo?

Reply:
Yes, use it to determin whether the prefetch operation is needed.
Thanks for your review @ramkrish86